Assets/Scripts/PlayerController/PlayerMovement.cs

```diff
@@ -67,10 +71,38 @@ void Update()
 
         HandleGravity();
         HandleMovement();
+        UpdateText();
 
         float t = Time.deltaTime;
         float t2 = t * t;
-        m_CharacterController.Move(new Vector3(m_Velocity.x, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, m_Velocity.z));
+
+        // m_canDoubleSpeed
+        float xVelocity = m_Velocity.x;
+        float yVelocity = m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2; 
+        float zVelocity = m_Velocity.z;
+
+        if (m_canDoubleSpeed)
+        {
+            xVelocity *= 2.0f;
+            zVelocity *= 2.0f;
+        }
+
+        if (m_canDoubleJump) 
+        {
+            if (yVelocity < 0) 
+            {
+                yVelocity *= 0.5f;
+            } else {
+                yVelocity *= 2.0f;
+            }
+        }
```

Comment:
Consider creating `velocityModifier`s and a `PowerupManager` that sets the modifiers instead. Otherwise as we add more and more modifiers this code will become very messy.

Reply:
Yea true that HAHAHA wanna get the mechanics right first, 

then OOP can do once the mechanics are settled